modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js

```diff
@@ -158,37 +128,46 @@ export default function CartQuickAdd() {
 	const handleAddToCartClick = () => {
 		const readySKUs = selectedSKUs.map((selectedSKU) => {
 			const {
+				discontinued,
 				id,
 				productConfiguration,
-				replacementSkuId,
+				replacementSku,
 				sku,
 				urls,
 			} = selectedSKU;
 
-			if (replacementSkuId) {
-				const replacementSKU = availableSKUs.find(
-					(availableSKU) => availableSKU.id === replacementSkuId
-				);
+			if (discontinued) {
```

Comment:
Check if selectedSKU is unavailable and if replacementSKU exists

Reply:
`if ((selectedSKU.availability?.label !== 'available') && replacementSKU)`

Reply:
✅